homeassistant/components/elevenlabs/config_flow.py

```diff
@@ -0,0 +1,191 @@
+    """Get available voices and models as dicts."""
+    client = AsyncElevenLabs(api_key=api_key)
+    voices = (await client.voices.get_all()).voices
+    models = await client.models.get_all()
+    voices_dict = {
+        voice.voice_id: voice.name
+        for voice in sorted(voices, key=lambda v: v.name or "")
+        if voice.name
+    }
+    models_dict = {
+        model.model_id: model.name
+        for model in sorted(models, key=lambda m: m.name or "")
+        if model.name and model.can_do_text_to_speech
+    }
+    return voices_dict, models_dict
+
+
+class ElevenLabsConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for ElevenLabs text-to-speech."""
+
+    VERSION = 1
+
+    user_info: dict[str, Any] | None = None
+
+    # id -> name
+    voices: dict[str, str] = {}
+    models: dict[str, str] = {}
+
+    def _get_user_schema_authenticated(self) -> vol.Schema:
+        assert self.voices and self.models
+        first_voice_id = next(iter(self.voices.keys()))
+
+        return vol.Schema(
+            {
+                vol.Required(
+                    CONF_VOICE, description={"suggested_value": first_voice_id}
+                ): SelectSelector(
+                    SelectSelectorConfig(
+                        options=[
+                            SelectOptionDict(label=voice_name, value=voice_id)
+                            for voice_id, voice_name in self.voices.items()
+                        ]
+                    )
+                ),
+                vol.Required(
+                    CONF_MODEL, description={"suggested_value": DEFAULT_MODEL}
+                ): SelectSelector(
+                    SelectSelectorConfig(
+                        options=[
+                            SelectOptionDict(label=model_name, value=model_id)
+                            for model_id, model_name in self.models.items()
+                        ]
+                    )
+                ),
+            },
+        )
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle the initial step."""
+        errors = {}
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA_NO_AUTH
+            )
+        # Validate auth, get voices
+        try:
+            self.voices, self.models = await get_voices_models(user_input[CONF_API_KEY])
+        except ApiError:
+            errors["base"] = "invalid_api_key"
+        if errors:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA_NO_AUTH, errors=errors
+            )
+        # Set model id
+        user_input.setdefault(CONF_MODEL, DEFAULT_MODEL)
+        self.user_info = user_input
+        return self.async_create_entry(
+            title=f"ElevenLabs {self.models[user_input[CONF_MODEL]]}", data=user_input
```

Comment:
No need to add `ElevenLabs` to title, it's already inside ElevenLabs integration.
```suggestion
            title=self.models[user_input[CONF_MODEL]], data=user_input
```

Reply:
Agreed, changed it.